server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1757,6 +1840,52 @@ private void updateWithLbRules(final DomainRouterJoinVO routerJoinVO, final Stri
         }
     }
 
+    private void updateLbValues(final HashMap<String, String> lbConfigsMap, StringBuilder loadBalancingData) {
+        String lbMaxConn = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbMaxConn.key(), null);
+        String lbFullConn = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbFullConn.key(), null);
+        String lbTimeoutConnect = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbTimeoutConnect.key(), null);
+        String lbTimeoutServer = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbTimeoutServer.key(), null);
+        String lbTimeoutClient = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbTimeoutClient.key(), null);
+        String lbBackendHttps = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbBackendHttps.key(), null);
+        String lbHttp2 = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbHttp2.key(), null);
+
+        // Process lb.server values
+        String serverMaxconn = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbServerMaxConn.key(), null);
+        String serverMinconn = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbServerMinConn.key(), null);
+        String serverMaxqueue = lbConfigsMap.getOrDefault(LoadBalancerConfigKey.LbServerMaxQueue.key(), null);
+
+        if (lbMaxConn != null) {
+            loadBalancingData.append(",lb.maxconn=").append(lbMaxConn);
+        }
+        if (lbFullConn != null) {
+            loadBalancingData.append(",lb.fullconn=").append(lbFullConn);
+        }
+        if (lbTimeoutConnect != null) {
+            loadBalancingData.append(",lb.timeout.connect=").append(lbTimeoutConnect);
+        }
+        if (lbTimeoutServer != null) {
+            loadBalancingData.append(",lb.timeout.server=").append(lbTimeoutServer);
+        }
+        if (lbTimeoutClient != null) {
+            loadBalancingData.append(",lb.timeout.client=").append(lbTimeoutClient);
+        }
+        if (lbBackendHttps != null) {
+            loadBalancingData.append(",lb.backend.https=").append(lbBackendHttps);
+        }
+        if (lbHttp2 != null) {
+            loadBalancingData.append(",http2=").append(lbHttp2);
+        }
+        if (serverMaxconn != null) {
+            loadBalancingData.append(",server.maxconn=").append(serverMaxconn);
+        }
+        if (serverMinconn != null) {
+            loadBalancingData.append(",server.minconn=").append(serverMinconn);
+        }
+        if (serverMaxqueue != null) {
+            loadBalancingData.append(",server.maxqueue=").append(serverMaxqueue);
+        }
```

Comment:
This code seems repeated, but changing parameters; We could extract it to a method.